File: src/main/java/kr/nyamnyam/model/domain/PostModel.java
package kr.nyamnyam.model.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;
import java.util.List;

@Component
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostModel {
    private Long id;
    private String content;
    private Long taste;
    private Long clean;
    private Long service;
    private LocalDateTime entryDate;
    private LocalDateTime modifyDate;

    private String userId;
    private String nickname;
    private Long restaurantId;

    private List<ImageModel> images;

    private double averageRating;
    private List<String> tags;

    public double averageRating() {
        return (this.taste + this.clean + this.service) / 3.0;
    }

}

File: src/app/model/post.model.ts
//src/app/model/post.model.ts
export interface PostModel {
  id: number;  
  taste: number;
  clean: number;
  service: number;
  content: string;
  entryDate: string;
  modifyDate: string;
  averageRating: number;
  tags?: string[]; 
  images: ImageModel[];
  restaurantId?: number;
  userId?: number;
  nickname?: string;
}

export const initialPost: PostModel = {
  id: 0,  
  taste: 0,
  clean: 0,
  service:0,
  content: '',
  entryDate: '',
  modifyDate: '',
  averageRating: 0,
  tags: [], 
  images: [],
  restaurantId: 0,
  userId: 0,
  nickname:''
}

